### PR TITLE
Invoice ln

### PR DESCRIPTION
**Pull request by lnchambers**

### CODE REVIEW

File: lib/sales_analyst.rb [hunk 1]
```diff
@@ -19,15 +19,13 @@ def mean_calculation_merchant(merchant)
...
-    end
-    calculate_stdev(mean)
+    calculate_stdev(merchant_mean)
   end
 
   def merchants_with_high_item_count
+    merchant_stdev = average_items_per_merchant_standard_deviation
     all_merchants.values.find_all do |merchant|
-      merchant.items.count > (average_items_per_merchant_standard_deviation * 2)
+      merchant.items.count > (merchant_stdev * 2)
```
Comment (lnchambers) [inline lib/sales_analyst.rb:28]: See this? We were running the STDev every time that we pulled one object out.

File: lib/sales_analyst.rb [hunk 2]
```diff
@@ -36,9 +34,10 @@ def average_item_price_standard_deviation
   end
 
   def golden_items
-    mean = all_items_price / total_items
+    item_stdev = calculate_item_stdev
+    item_mean = item_price_mean
     all_items.values.find_all do |item|
-      item.unit_price > (mean + average_item_price_standard_deviation * 2)
+      item.unit_price > (item_mean + item_stdev * 2)
```
Comment (lnchambers) [inline lib/sales_analyst.rb:40]: Same thing here.

File: test/sales_analyst_test.rb [hunk 3]
```diff
@@ -81,10 +86,11 @@ def test_average_item_price_standard_deviation
     sales_engine = SalesEngine.from_csv({
       :items     => "./test/fixtures/item_fixture.csv",
       :merchants => "./test/fixtures/merchant_fixture.csv",
+      :invoices  => "./test/fixtures/invoice_fixture.csv"
```
Comment (lnchambers) [inline test/sales_analyst_test.rb:89]: Need to fix these syntactically

File: lib/sales_analyst.rb (hunk 1 above)
Comment (zpalmquist) [reply lib/sales_analyst.rb:28]: Right! This was what I was getting at but wasn't sure how to implement. Killer.

File: test/sales_analyst_test.rb (hunk 3 above)
Comment (zpalmquist) [reply test/sales_analyst_test.rb:89]: Going to use the setup that I made on my tests and cut all that out. You'll see in the next merge.

**Review by zpalmquist — APPROVED** [review]
Hell yes! I think we will have to do the beer beer beer beer beer beer beer beer thing after this is all done.

**Pull request merged**